src/config/backends/extlinux-file.ts

```diff
@@ -0,0 +1,166 @@
+import * as _ from 'lodash';
+
+import { ConfigOptions } from './backend';
+import {
+	ExtLinuxParseError,
+	AppendDirectiveError,
+	FDTDirectiveError,
+} from '../../lib/errors';
+
+export interface ExtlinuxFile {
+	globals: Directive;
+	labels: Label;
+}
+
+export interface Label {
+	[labelName: string]: Directive;
+}
+
+export interface Directive {
+	[directive: string]: string;
+}
+
+/**
+ * ConfigurableDirective
+ *
+ * This class abstraction is the blueprint used to create new directives in extlinux
+ * that we would want to be able to parse (get the value) and generate (create a value).
+ *
+ */
+export abstract class ConfigurableDirective {
+	// Parses the values for this directive
+	public abstract parse(directives: Directive): ConfigOptions;
+	// Return the value to be set for this directive using the provided ConfigOptions
+	public abstract generate(opts: ConfigOptions, existingValue?: string): string;
+}
+
+/**
+ * AppendDirective
+ *
+ * Add one or more options to the kernel command line.
+ *
+ */
+export class AppendDirective extends ConfigurableDirective {
+	private supportedConfigValues: string[];
+
+	public constructor(supportedConfigValues: string[]) {
+		super();
+		this.supportedConfigValues = supportedConfigValues;
+	}
+
+	/**
+	 * Parses a APPEND directive string into a ConfigOptions
+	 *
+	 * Example:
+	 * 	parse({ APPEND: "ro rootwait isolcpus=0,4" })
+	 * 	-> { 'ro': '', 'rootwait': '', 'isolcpus': '0,4' }
+	 *
+	 */
+	public parse(directives: Directive): ConfigOptions {
+		// Check that there is an APPEND directive to parse
+		if (directives.APPEND == null) {
+			throw new ExtLinuxParseError(
+				'Could not find APPEND directive in default extlinux.conf boot entry',
+			);
+		}
+		// Parse all the key and values into ConfigOptions
+		return directives.APPEND.split(' ').reduce(
+			(configOpts: ConfigOptions, appendValue: string) => {
+				// Break this append config into key and value
+				const [KEY, VALUE = '', ...more] = appendValue.split('=');
+				if (!KEY) {
+					return configOpts; // No value to set so return
+				} else if (more.length > 0) {
```

Comment:
You can limit to a max of 3 returned and also avoid building up the whole array of junk results by doing:
```suggestion
				const [KEY, VALUE = '', more] = appendValue.split('=', 3);
				if (!KEY) {
					return configOpts; // No value to set so return
				} else if (more != null) {
```
since `more` will either be a string or nullish:
```ts
'a=b='.split('=') // ["a", "b", ""]
'a=b'.split('=') // ["a", "b"]

'a=b==='.split('=') // ["a", "b", "", "", ""]
'a=b==='.split('=', 3) // ["a", "b", ""]

Reply:
I like this philosophy of specifying the exact number of splits to return. I'll be sure to use it more. added.